operator/controllers/csiscaleoperator_controller.go

```diff
@@ -405,16 +405,16 @@ func (r *CSIScaleOperatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
 					// Daemon set should not update when secret is updated but not the data.
 					// e.g Secret type, resource version etc is modified.
 					// TODO: Update only those daemon set which are in the same namespace as the secret that triggered the event.
-					for _, daemonSet := range CSIDaemonListFunc() {
+					daemonSets := CSIDaemonListFunc()
+					for i := range CSIDaemonListFunc() {
 						logger.Info("Secrets were modified. Daemon Set will be updated. Restarting node specific pods.")
-						err = r.rolloutRestartNode(&daemonSet)
+						err = r.rolloutRestartNode(&daemonSets[i])
```

Comment:
May be I am not getting how this will work.. With the change 'i' will be of type DaemonSet, then how it can be used for indexing the list 'daemonSets' ?

Reply:
`range` keyword returns 2 elements:
1. index
2. item in the array

by using:
```
for i := range CSIDaemonListFunc() {

}
```
we are ignoring the 2nd element, which is as good as:
```
for i, _ := range CSIDaemonListFunc() {

}

So we are able to use index i to access elements in the array.